pkg/platform/kube/deployer.go

```diff
@@ -250,7 +250,34 @@ func (d *deployer) getFunctionPodLogs(namespace string, name string) (string, st
 			// close the stream
 			logsRequest.Close() // nolint: errcheck
 		}
+
+		podWarningEvents, err := d.getFunctionPodWarningEvents(namespace, pod.Name)
+		if err != nil {
+			podLogsMessage += "Failed to get pod events\n"
+		} else if briefErrorMessage == "" && podWarningEvents != "" {
+
+			// if there is no brief error message and there are warning events - add them
+			podLogsMessage += "\n* Warning events:\n" + podWarningEvents
+			briefErrorMessage += podWarningEvents
+		}
 	}
 
 	return podLogsMessage, briefErrorMessage
 }
+
+func (d *deployer) getFunctionPodWarningEvents(namespace string, podName string) (string, error) {
+	eventList, err := d.consumer.kubeClientSet.CoreV1().Events(namespace).List(meta_v1.ListOptions{})
+	if err != nil {
+		d.logger.InfoWith("Failed to get pod events", "err", err)
```

Comment:
hmm, return the error with `errors.New("Failed to get pod warning events")`
and from its callee, simply append the log message (like you did above with `podLogsMessage += "Failed to read logs: " + getLogsErr.Error() + "\n"`